src/java/main/org/apache/zookeeper/server/quorum/QuorumPeerConfig.java

```diff
@@ -270,6 +271,8 @@ public void parseProperties(Properties zkProp)
                 quorumServicePrincipal = value;
             } else if (key.equals("quorum.cnxn.threads.size")) {
                 quorumCnxnThreadsSize = Integer.parseInt(value);
+            } else if (key.equals("tcpKeepAlive")) {
+                tcpKeepAlive = Boolean.parseBoolean(value);
```

Comment:
As previously mentioned you can just let it fall through and query the system property when we need set the sock option. 
If we do need parse it, then we need handle potential exceptions on illegal values here.